src/confd/src/infix-containers.c

```diff
@@ -206,24 +205,18 @@ static int add(const char *name, struct lyd_node *cif)
 	fchmod(fileno(fp), 0700);
 	fclose(fp);
 
-	/* Enable, or update, container -- both trigger setup script. */
+	/* Enable, or update, container -- both trigger container setup. */
 	systemf("initctl -bnq enable container@%s.conf", name);
 	systemf("initctl -bnq touch container@%s.conf", name);
 
 	return 0;
 }
 
+/* Remove container setup script and disable currently running instance */
 static int del(const char *name)
 {
-	char fn[strlen(_PATH_CONT) + strlen(name) + 10];
-
-	/* Remove container setup script */
-	snprintf(fn, sizeof(fn), "%s/%s.sh", _PATH_CONT, name);
-	erase(fn);
-
-	/* Stop and schedule for deletion */
-	systemf("initctl -bnq stop container:%s", name);
-	writesf(name, "a", "%s", _PATH_INBOX);
+	erasef("%s/%s.sh", _PATH_CONT, name);
+	systemf("initctl -bnq disable container@%s.conf", name);
```

Comment:
Is there any risk of a race between `erasef()` wiping out files that podman still expects to be available until the container has been stopped? I.e., should these two lines be reversed? Letting `podman` stop the container before cleaning up the filesystem.

Do we expect that `erasef()` will do any work here? If so, could/should it be moved to the cleanup hook of the service?

Reply:
No risk of that, the `container` wrapper script and any container create/recreate/upgrade script are independent.

Yes, I think `erasef()` fits here because we create it here, in the function just prior.  

But good points nonetheless, I'll add a comment to the code about their relationship.  Because I'll likely forget myself soon :laughing: